apm-sniffer/apm-sdk-plugin/httpClient-4.x-plugin/src/main/java/org/skywalking/apm/plugin/httpClient/v4/HttpClientExecuteInterceptor.java

```diff
@@ -12,68 +12,47 @@
 import org.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.skywalking.apm.network.trace.component.ComponentsDefine;
 
 /**
  * {@link HttpClientExecuteInterceptor} transport the trace context by call {@link HttpRequest#setHeader(Header)},
- * The current span tag the {@link Tags#ERROR} if {@link StatusLine#getStatusCode()} is not equals 200.
+ * The {@link Tags#STATUS_CODE} will be set if {@link StatusLine#getStatusCode()} is not equals 200.
  *
  * @author zhangxin
  */
 public class HttpClientExecuteInterceptor implements InstanceMethodsAroundInterceptor {
-    private static final String COMPONENT_NAME = "HttpClient";
 
-    @Override
-    public void beforeMethod(EnhancedClassInstanceContext context,
-        InstanceMethodInvokeContext interceptorContext, MethodInterceptResult result) {
-        Object[] allArguments = interceptorContext.allArguments();
+    @Override public void beforeMethod(EnhancedInstance objInst, String methodName, Object[] allArguments,
+        Class<?>[] argumentsTypes, MethodInterceptResult result) throws Throwable {
         if (allArguments[0] == null || allArguments[1] == null) {
             // illegal args, can't trace. ignore.
             return;
         }
-        HttpHost httpHost = (HttpHost)allArguments[0];
+        final HttpHost httpHost = (HttpHost)allArguments[0];
         HttpRequest httpRequest = (HttpRequest)allArguments[1];
-        AbstractSpan span = createSpan(httpRequest);
-        span.setPeerHost(httpHost.getHostName());
-        span.setPort(httpHost.getPort());
-        Tags.SPAN_KIND.set(span, Tags.SPAN_KIND_CLIENT);
-        Tags.COMPONENT.set(span, COMPONENT_NAME);
-        Tags.URL.set(span, generateURL(httpRequest));
-        Tags.SPAN_LAYER.asHttp(span);
-
-        ContextCarrier contextCarrier = new ContextCarrier();
-        ContextManager.inject(contextCarrier);
-        httpRequest.setHeader(Config.Plugin.Propagation.HEADER_NAME, contextCarrier.serialize());
-    }
-
-    /**
-     * Format request URL.
-     *
-     * @return request URL
-     */
-    private String generateURL(HttpRequest httpRequest) {
-        return httpRequest.getRequestLine().getUri();
-    }
-
-    /**
-     * Create span.
-     */
-    private AbstractSpan createSpan(HttpRequest httpRequest) {
+        final ContextCarrier contextCarrier = new ContextCarrier();
         AbstractSpan span;
+        String remotePeer = httpHost.getHostName() + ":" + httpHost.getPort();
         try {
             URL url = new URL(httpRequest.getRequestLine().getUri());
-            span = ContextManager.createSpan(url.getPath());
+            span = ContextManager.createExitSpan(url.getPath(), contextCarrier, remotePeer);
         } catch (MalformedURLException e) {
-            span = ContextManager.createSpan(httpRequest.getRequestLine().getUri());
+            span = ContextManager.createExitSpan(httpRequest.getRequestLine().getUri(), contextCarrier, remotePeer);
```

Comment:
Why do the nearly same thing in `catch` block? What is the difference between URL and URI?